File: agent/bindings/csharp/WorkspaceEditEntryMetadata.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class WorkspaceEditEntryMetadata
  {
    [JsonProperty(PropertyName = "needsConfirmation")]
    public bool NeedsConfirmation { get; set; }
    [JsonProperty(PropertyName = "label")]
    public string Label { get; set; }
    [JsonProperty(PropertyName = "description")]
    public string Description { get; set; }
    [JsonProperty(PropertyName = "iconPath")]
    public Uri IconPath { get; set; }
  }
}

File: agent/bindings/kotlin/lib/src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/WorkspaceEditEntryMetadata.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class WorkspaceEditEntryMetadata(
  val needsConfirmation: Boolean,
  val label: String,
  val description: String? = null,
  val iconPath: Uri,
)

